tests/moe/test_alphamoe_nvfp4_sm100.py

```
@@ -0,0 +1,538 @@
+    with pytest.raises(ValueError, match="16-byte aligned"):
+        alphamoe_nvfp4_aligned_moe(*misaligned_args)
+
+    misaligned_w1_storage = torch.empty(
+        case["w1"].numel() + 1, dtype=torch.uint8, device="cuda"
+    )
+    misaligned_w1 = misaligned_w1_storage[1:].view_as(case["w1"])
+    assert misaligned_w1.is_contiguous()
+    assert misaligned_w1.data_ptr() % 16 != 0
+    misaligned_w1_args = list(args)
+    misaligned_w1_args[2] = misaligned_w1
+    with pytest.raises(ValueError, match="gemm1_weights.*16-byte aligned"):
```

Ruff (0.16.0)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)